### PR TITLE
@orta => X callback url

### PR DESCRIPTION
**Pull request by 1aurabrown**
This adds a pod for handling x-callback-url according to the protocol's conventions. right now, it just handles opening a url as a view and pushing it, which is nothing new. next i'll wire up the back button to take you back to whatever place you came from.

the incoming url would be structured like this:
`artsy-x-callback-url://x-callback-url/open?url=some/artsy/url`

this is how you'er supposed to do it according to the x-callback-url specs. `open` is an action that takes some parameters, in the case, `url`. this should also work with a path.

### CODE REVIEW

**Comment by alloy** [comment]
What’s this going to be used for?

**Comment by alloy** [comment]
Btw, usually a custom (unstandardised) scheme is prefixed with `x-`, e.g. `x-artsy`.

File: Artsy/Classes/ARAppDelegate.m [hunk 1]
```diff
@@ -140,6 +145,23 @@ - (void)showTrialOnboardingWithState:(enum ARInitialOnboardingState)state andCon
...
+        NSString *url = inputParameters[@"url"];
+        if (!url) {
+            failure();
+            return;
+        }
+
+        UIViewController *viewController = [ARSwitchBoard.sharedInstance loadURL:url];
+        // Do things to set up back button.
+        if (viewController) {
+            [self.viewControllerDelegate textView:self shouldOpenViewController:viewController];
```
Comment (1aurabrown) [inline Artsy/Classes/ARAppDelegate.m]: this is a bad copy paste, will fix.

**Comment by 1aurabrown** [comment]
@alloy this will be used to allow other applications to launch artsy and provide a callback url, so that our back button will take you back to the source app when you hit the back button

**Comment by 1aurabrown** [comment]
i can rename the scheme.
google chrome uses `googlechrome-x-callback://` for their x-callback-url scheme fwiw

**Comment by 1aurabrown** [comment]
it seems some apps don't use a separate scheme for x-callback-url, but the protocol dictates that you should

**Comment by 1aurabrown** [comment]
let's hold off on this, i noticed some sloppy mistakes and i'm adding more work.

**Comment by orta** [comment]
@alloy x-callback is an unofficial app standard for sending messages between apps, we've been asked about support before from other devs and not been able to do it "right" - http://x-callback-url.com

**Comment by alloy** [comment]
Yeah I get what the thing is, I'm just wondering what it is that Artsy will provide in terms of interop :)

**Comment by alloy** [comment]
@1aurabrown Yeah there's no standard that defines what unstandardised schemes should look like, so whatever works 🙅

**Comment by 1aurabrown** [comment]
@orta i think i'm ready for you to take a look at this.

File: Artsy/Classes/Utils/ARBackButtonCallbackManager.m [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    _viewController = viewController;
+    _backBlock = backBlock;
+
+    return self;
+}
+
+- (BOOL)canHandleBackForViewController:(UIViewController *)viewController;
+{
+    return self.viewController != nil &&
+    self.viewController == viewController;
```
Comment (orta) [inline Artsy/Classes/Utils/ARBackButtonCallbackManager.m]: looks weird

**Comment by orta** [comment]
Fails are all in ARLoginViewController, merging.

**Comment by 1aurabrown** [comment]
yay

**Comment by alloy** [comment]
This was missing a CHANGELOG, I added on [here](3f6abbdd27f5f0934410e02cfd6733b8c4f4d4bb), please review it.

**Pull request merged**